### DISCUSSION TITLE
Testes automatizados para react Native

### DISCUSSION THREAD

**Discussion by fbellato** [frontendbr/forum#1157] [Perguntas e Respostas]
Fala galera, meu time está começando a desenvolver um novo app em React Native integrado com o expo, estamos em duvida ainda sobre a ferramenta utilizada para testes automatizados.
O Detox pareceu uma ferramenta boa, mas ele ainda não tem integração com expo+Android, por enquanto só expo+ios.
Estamos pensando então em utilizar o Appium e rodar os testes no travis ou em alguma device farm, o problema é que o travis demoraria muito para baixar e montar todos ambientes e então rodar os testes em modo headless, atrasando muito todo o processo de deploy. Alguém tem alguma experiencia com algo parecido ou alguma ideia pra nos ajudar?
Valeu

**Comment by felipefialho** [comment]
@fdaciuk

**Comment by raisiqueira** [comment]
@fbellato pq estão começando com Expo? É um prótotipo ou algo do tipo?

**Comment by luisfmsouza** [comment]
Fala @fbellato, blz?

Aqui na Magnetis nós estamos usando o [Cavy](https://github.com/pixielabs/cavy) pra fazer nossos testes de integração, e tem funcionado muito bem! No passado nós testamos o Detox, mas desistimos justamente pela falta de suporte ao Android (no futuro, quando houver a integração, esperamos poder usar ele pra algo).

Pra rodar os nossos testes de forma automatizada, nós utilizamos um CI chamado [Bitrise](https://www.bitrise.io/). Nele conseguimos provisionar um device de cada OS e rodar os testes nas nossas builds do Github, por exemplo.

Um colega do trabalho [escreveu esse post](https://medium.com/magnetis-backstage/react-native-e2e-testing-with-cavy-1f1d5be1d3be), dando mais detalhes sobre o que fizemos com o Cavy. Recomendo a leitura e assistir o vídeo que tem no post.

Qualquer nova dúvida, pode mandar ai!

**Reply by Keikonichi** [reply]
essa página não aparece mais.  Error 522 Ray ID: 6b44336bad246abd • 2021-11-26 15:37:56 UTC
Connection timed out
dá erro

**Comment by regianer** [comment]
Olá, vcs tem mais experiencia para compartilhar do react native no android e ios? fizeram testes UI? Chegaram a tentar appium?
Eu estou passando algumas dificuldades com isso

**Comment by luisfmsouza** [comment]
@regianer recentemente li esse post e achei muito interessante: https://www.polidea.com/blog/how-to-apply-ui-test-automation-in-react-native-apps
